src/ipfs-core/index.js

```diff
@@ -1,99 +1,79 @@
-// var ipfsAPIclt = require('ipfs-api')
-// var extend = require('extend')
-// var PeerId = require('peer-id')
-// var PeerInfo = require('peer-info')
-var config = require('./config')
-var IPFSRepo = require('ipfs-repo')
+'use strict'
+
+const config = require('./config')
+const IPFSRepo = require('ipfs-repo')
 
 exports = module.exports = IPFS
 exports.config = config
 
 function IPFS () {
-  var self = this
-
-  if (!(self instanceof IPFS)) {
+  if (!(this instanceof IPFS)) {
```

Comment:
I was not sure how classes worked underneath, was afraid that it would cause the same problem of method allocation during object initialisation and take a lot of unnecessary memory, but after checking, it seems there is no reason why to be concerned since it moves all the methods to .prototype of class.